packages/query-async-storage-persister/src/__tests__/asyncThrottle.test.ts

```diff
@@ -105,11 +106,10 @@ describe('asyncThrottle', () => {
     expect(mockFunc.mock.calls[1]?.[0]).toBe(2)
   })
 
-  test('"onError" should be called when "func" throw error', (done) => {
+  test('"onError" should be called when "func" throw error', () => {
     const err = new Error('error')
     const handleError = (e: unknown) => {
       expect(e).toBe(err)
-      done()
```

Comment:
how does vitest know to wait for this expectation to occur ?

Reply:
I believe vitest will wait for any unresolved promises created by the test's callback closure to resolve before continuing. I affirmed this assumption by adding a false assertion (`expect(true).toBe(false)`) where the `done()` was and ensuring the test fails).

If we want we can make the test callback `async` and explicitely `await` the last line of this test if we think this behavior might confuse folks!

Reply:
yeah that looks good 👍